test/unit/atomic/src/AtomicFunctors.hpp

```diff
@@ -0,0 +1,143 @@
+/* Copyright 2022 Axel Huebl, Benjamin Worpitz, Matthias Werner, Jan Stephan, Bernhard Manfred Gruber,
+ * Antonio Di Pilato
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file This file provides functors where a atomic function and the corresponding operation availble within one
+//! object. This allwos testing of the atomic function and atomicOp<Op>() without the usage of function pointers whcih
+//! is not suppoted by all backends.
+
+#include <alpaka/atomic/Op.hpp>
+#include <alpaka/atomic/Traits.hpp>
+
+#include <utility>
+
+namespace alpaka::test::unit::atomic
+{
+    struct Add
+    {
+        using Op = alpaka::AtomicAdd;
+
+        template<typename... TArgs>
+        static ALPAKA_FN_ACC auto atomic(TArgs&&... args)
```

Comment:
@jkelling found out that this variadic template usage is the root of the nvhpc error (invalid memory access)